provider/pkg/await/deployment.go

```diff
@@ -382,14 +454,38 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 
 	// Do nothing if this is not the Deployment we're waiting for.
 	if deployment.GetName() != inputDeploymentName {
+		logger.V(9).Infof("Watch event NOT actually for deployment: %s", dia.deployment.GetName())
 		return
 	}
 
+	logger.V(9).Infof("Watch event for deployment: %s", dia.deployment.GetName())
 	// Mark the rollout as incomplete if it's deleted.
 	if event.Type == watch.Deleted {
+		logger.V(9).Infof("Watch event for deployment: %s - DELETED", dia.deployment.GetName())
 		return
 	}
 
+	newGeneration := deployment.GetAnnotations()[revision]
+	if dia.deployment != nil {
+		if dia.deployment.GetResourceVersion() > deployment.GetResourceVersion() {
+			logger.V(3).Infof("Deployment %s received an older event. Known version: %q, received: %q",
+				dia.deployment.GetName(),
+				dia.deployment.GetResourceVersion(),
+				deployment.GetResourceVersion())
+			return
+		}
+
+		currentGeneration := dia.deployment.GetAnnotations()[revision]
```

Comment:
Is this annotation guaranteed to be present? If it's not set, the return value will be `""`. (I noticed that this was already present in other logic in this awaiter, but it might be good to be defensive.)